lib/charms/tls_certificates_interface/v2/tls_certificates.py

```diff
@@ -897,6 +897,22 @@ def __init__(self, charm: CharmBase, relationship_name: str):
         self.charm = charm
         self.relationship_name = relationship_name
 
+    def _load_relation_data(self, raw_relation_data: dict) -> dict:
+        """Loads relation data from the relation data bag.
+
+        Json loads all data.
+
+        Args:
+            raw_relation_data (dict): Relation data from the databag
+
+        Returns:
+            dict: Relation data in dict
+        """
+        # If unit is not leader, it does not try to reach relation data.
+        if not self.model.unit.is_leader():
+            return {}
+        return _load_relation_data(raw_relation_data)
```

Comment:
is it calling itself? or do you mean to have a different function here?

Reply:
oh, never mind, I see it's calling the global function.

Reply:
_load_relation_data is the original function here. Above is the method under TLSCertificatesProvidesV2 that calls the function.